Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -84,103 +84,140 @@ namespace ROS2
         }
     }
 
-    void URDFPrefabMaker::BuildAssetsForLink(const sdf::Link* link)
+    URDFPrefabMaker::CreatePrefabTemplateResult URDFPrefabMaker::CreatePrefabTemplateFromUrdfOrSdf()
     {
-        m_collidersMaker.BuildColliders(link);
+        {
+            AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+            m_status.clear();
+        }
+
+        if (!ContainsModel())
+        {
+            return AZ::Failure(AZStd::string("URDF/SDF doesn't contain any models."));
+        }
```

Comment:
Is this check necessary? Can't we have a valid <world> without a <model>? 

Admittedly, it would probably be a pretty useless conversion, so maybe it's ok to check this...

Reply:
Technically the check isn't necessary, It would just result in a prefab with no entities in it.

Reply:
Once we support `<light>`, we could technically have `<world><light> ... </light></world>` that would have an entity but no models I guess. Still seems like a pretty silly and contrived example though, so I'm pretty ambivalent on the check, just pointing it out.

Reply:
Well at that point we can update the function to check if the SDF contains a model or a light.